src/cfnlint/rules/resources/properties/AllowedValues.py

```diff
@@ -0,0 +1,149 @@
+                    resourcetype = str.format('{0}.{1}', parenttype, proptype)
+            else:
+                resourcetype = str.format('{0}.{1}', parenttype, proptype)
+
+        resourcespec = specs[resourcetype]['Properties']
+        if not isinstance(text, dict):
+            # Covered with Properties
+            return matches
+
+        resource_objects = list()
+        base_object_properties = {}
+        for key, value in text.items():
+            if key not in cfnlint.helpers.CONDITION_FUNCTIONS:
+                base_object_properties[key] = value
+        condition_found = False
+        for key, value in text.items():
+            if key in cfnlint.helpers.CONDITION_FUNCTIONS:
+                condition_found = True
+                cond_values = self.cfn.get_condition_values(value)
+                for cond_value in cond_values:
+                    if isinstance(cond_value['Value'], dict):
+                        append_object = {}
+                        append_object['Path'] = tree[:] + [key] + cond_value['Path']
+                        for sub_key, sub_value in cond_value['Value'].items():
+                            append_object[sub_key] = sub_value
+
+                        append_object['Value'] = append_object
+                        append_object['Value'].update(base_object_properties)
+                        resource_objects.append(append_object)
+
+        if not condition_found:
+            resource_objects.append({
+                'Path': tree[:],
+                'Value': base_object_properties
+            })
+
+        for resource_object in resource_objects:
+            path = resource_object.get('Path')
+            value = resource_object.get('Value')
+            for prop in resourcespec:
+
+                allowed_values = resourcespec[prop].get('AllowedValues')
+                # Check if allowed values are specified
+                if allowed_values:
+                    # Check if there's a value at all. Required is handled by
+                    # the required properties rule (E3003)
+                    if prop in value:
+
+                        if value[prop] not in allowed_values:
+                            message = 'Invalid property value "{0}" specified for {1}. Allowed value(s): {2}'
+                            spec_path = path + [prop]
+                            matches.append(RuleMatch(spec_path, message.format(
+                                value[prop],
+                                '/'.join(map(str, spec_path)),
+                                ', '.join(map(str, allowed_values))
+                            )))
+
+            # For all specified properties, check all nested properties
+            for prop in value:
+                proptree = path[:]
+                proptree.append(prop)
+                if prop in resourcespec:
+                    if 'Type' in resourcespec[prop]:
+                        if resourcespec[prop]['Type'] == 'List':
```

Comment:
Would this ever actually get hit?  I'm not sure a Type List is in this spec.

Reply:
Need to add valid support like Ref's etc also in there anyway. I'll continue on changing this POC

Reply:
More like: If it's a ref to a parameter, check the default values from the parameter

Reply:
Got it.  Checking the default value would be great.

The other one but probably a different rule is if it had AllowedValues that those individual allowed values are within the spec.

Reply:
But that might be bit "much" for now (mvp)

Reply:
Yep agreed. Different rule diffferent day